src/server/viper.ts

```diff
@@ -38,149 +35,161 @@ try {
         },
     ).on('error', console.error);
 
-    const socket = connect(serverPort);
-    connection = rpc.createMessageConnection(
-        new rpc.SocketMessageReader(socket),
-        new rpc.SocketMessageWriter(socket),
-    );
-    connection.listen();
+    server.stdout.on('data', (data) => {
+        if (connection != undefined) return;
+        const s = (data as Buffer).toString()
+        console.log(s);
+        const m = s.match(/<ViperServerPort:([0-9]+)>/);
+        if (!m) {
+            return
+        } else {
+            const port = Number(m[1]);
+            const socket = connect(port);
+            connection = rpc.createMessageConnection(
+                new rpc.SocketMessageReader(socket),
+                new rpc.SocketMessageWriter(socket),
+            );
+            connection.listen();
 
-    console.log('Listening to Viper LSP');
+            console.log('Listening to Viper LSP (port: %s)', port);
```

Comment:
what the fresh hell, why doesn't this format properly? I get
```
Listening to Viper LSP (port: %s) 52156
```

Reply:
`console.log` is strange (since it adds spaces between arguments except in specific situations). I would use [template strings](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) in this situation.

Reply:
Okay I am using this too, now! Thanks!